patches/tModLoader/Terraria.ModLoader.Core/LogArchiver.cs

```diff
@@ -13,8 +13,22 @@ namespace Terraria.ModLoader.Core
 	internal static class LogArchiver
 	{
 		internal static void ArchiveLogs() {
-			foreach (var logFile in Directory.GetFiles(Logging.LogDir, "*.old*"))
-				Archive(logFile, Path.GetFileNameWithoutExtension(logFile));
+			if (!Directory.Exists(Logging.LogArchiveDir)) {
+				string[] existingOldLogs = Directory.GetFiles(Logging.LogDir, "*.zip");
+
+				for (int i = 0; i < existingOldLogs.Length; i++) {
+					try {
+						File.Delete(existingOldLogs[i]);
+					}
+					catch (IOException) { }
+				}
+
+				Directory.CreateDirectory(Logging.LogArchiveDir);
+			}
+
+			foreach (string logFile in Directory.GetFiles(Logging.LogDir, "*.old*")) {
+				Archive(logFile, Path.GetFileNameWithoutExtension(logFile));

```

Comment:
This is moving the files to the new old folder, right?

Reply:
This checks if the Old folder is existant, and if not, deletes the old archives logs and creates the Old folder

Reply:
yes